File: frontend/src/shared/types/agent.ts
export type AgentCitation = {
  title: string;
  source_type: string;
  path: string;
  updated_at: string;
  snippet: string;
};

export type ChatMessage = {
  role: "user" | "assistant" | string;
  content: string;
};

export type ChatResponse = {
  session_id: string;
  answer: string;
  citations: AgentCitation[];
  context_summary: string;
  suggested_next_questions: string[];
  is_uncertain: boolean;
  retrieval_status: "matched" | "no_match" | string;
  ai_generated: boolean;
};

File: backend/app/schemas/agent.py
from pydantic import BaseModel, Field


class ChatRequest(BaseModel):
    message: str = Field(min_length=1)
    scenario: str = "general"


class Citation(BaseModel):
    title: str
    source_type: str
    snippet: str


class ChatResponse(BaseModel):
    answer: str
    citations: list[Citation] = Field(default_factory=list)
    ai_generated: bool = True
